#! /usr/bin/env python3.6
# -*- coding: utf-8 -*-
from dynamic_reconfigure.parameter_generator_catkin import *

"""
    动态参数客户端：
        1.导包；
        2.创建一个参数生成器；
        3.往生成器添加参数；
        4.配置节点，并退出
        aughor
"""
# 2.创建一个参数生成器；
gen = ParameterGenerator()
# 3.往生成器添加参数；
#add: (name: Any, paramtype: Any, level: Any, description: Any, 
# default: Any | None = None, 
# min: Any | None = None, 
# max: Any | None = None, edit_method: str = "") -> None

#Car parameter
gen.add("L", double_t,0,"length of Car",0.3,0.2,0.4)
gen.add("Lfw", double_t,0,"*min d when car is turning*", 0.55, 0.1, 2.0)
gen.add("lfw", double_t,0,"forward look ahead distance (m)",0.13,0.0,5.0)
gen.add("angle_gain_forward", double_t,0,"*gain of the turnning angle(to be tested)*",1.35, 0.0, 10.0)#这里的初始值可能会影响刚开启节点的时候舵机的角度？？
gen.add("base_speed_forward", double_t,0,"*base speed of car*", 0.5, 0.2, 2.0)

# gen.add("y_limit", double_t,0,"unknown parapm, don't changed it",-7.569,-10,0)
gen.add("AllowReverse",bool_t,0,"reverse or not",False)
gen.add("Lrv", double_t,0,"unknown parapm, don't changed it",0.4,0,10)
gen.add("lrv", double_t,0,"distance between front the center of car",0.0,-5.0,5.0)


#Controller parameter
gen.add("angle_gain_back", double_t,0,"*gain of the turnning angle(to be tested)*",1.0, 0.0, 3.0)
gen.add("base_speed_back", double_t,0,"*base speed of car*", 0.35, 0.2, 2.0)
gen.add("controller_freq", double_t,0,"controller's freqency(Hz)", 20, 5, 50)
gen.add("Vcmd", double_t,0,"*reference speed (m/s)*",0.7,0.5,1.5)
gen.add("gas_gain", double_t,0,"gas gain of car(to be tested)", 1.0, 0.5, 1.5)
gen.add("base_angle", double_t,0,"*base angle of car(to be tested)*", 0.0, -0.5, 0.5)
# # 4.配置节点，并退出。
# generate: (pkgname: Any, nodename: Any, name: Any) -> Any
#最后一个name和cfg文件的名字保持一致
exit(gen.generate("qingzhou_nav","L1_dynamic","L1_dynamic"))